- make tests and benchmarks conditional on special cmake variables, see https://github.com/tdlib/td/issues/1583

--- CMakeLists.txt.orig	2021-05-20 23:05:03 UTC
+++ CMakeLists.txt
@@ -189,9 +189,11 @@ add_subdirectory(sqlite)
 
 add_subdirectory(tddb)
 
+if (BUILD_TESTING)
 add_subdirectory(test)
+endif()
 
-if (NOT CMAKE_CROSSCOMPILING)
+if (FREEBSD_BUILD_BENCHMARKS AND NOT CMAKE_CROSSCOMPILING)
   add_subdirectory(benchmark)
 endif()
 
